### PR TITLE
 Blog for the Symplectic Integrator Project in Geant4

### PR DESCRIPTION
**Pull request by divyanshtiwari237**
This blog contains all the information about my work done till the Mid-Term evaluation period.

### CODE REVIEW

**Comment by divyanshtiwari237** [comment]
cc @jonapost for review

**Review by jonapost — COMMENTED** [review]
Line 32 "excellent performance in electromagnetic field" is loose. It is specific for EM fields. We chose it for its simplicity and wide use

**Comment by jonapost** [comment]
Good overview. Please make a small change

File: _gsocblogs/2022/blog_Geant4_DivyanshTiwari.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+project: Geant4
+title: Symplectic Integrators
+author: Divyansh Tiwari
+date: 27.07.2022
+year: 2022
+layout: blog_post
+logo: Geant4-logo.png
+intro: |
+  Implementing Symplectic Integrators
```
Comment (agheata) [inline _gsocblogs/2022/blog_Geant4_DivyanshTiwari.md:10]: The introduction must not be just a title, it should contain a sentence telling the readers what they are going to read about.

File: _gsocblogs/2022/blog_Geant4_DivyanshTiwari.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+layout: blog_post
+logo: Geant4-logo.png
+intro: |
+  Implementing Symplectic Integrators 
+---
+
+## Introduction
+
+Hello Everyone! This is Divyansh. I was accepted to take part in GSoC 2022 under CERN-HSF and I am contributing to Geant4. My project is about implementing symplectic integrators into the Geant4 codebase. Many applications require the use of integrator method that preserve evergy over the course of a simulation.
+Hence, the aim of this project is to implement multiple methods which do not accumulate errors in energy and phase space volume over a large number of integration steps.A famous class of methods which fulfill the above criteria are the symplectic integrators.
```
Comment (agheata) [inline _gsocblogs/2022/blog_Geant4_DivyanshTiwari.md:16]: You can just copy this verbatim as `intro` text above.

File: _gsocblogs/2022/blog_Geant4_DivyanshTiwari.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+## Coding Period - Kick Off
+
+Before starting work on the actual methods, it was important develop up a test bed application that could be used to test the steppers and drivers. Hence, I started off my coding period by implementing a geant4 application to track a particle in a geometry initialized with a magnetic field. This was really interesting and exciting for me as I had done my homework during the community bonding period. The test bed is comprised of a detector constructor class, a physics list class and a primary action generator class. The magnetic field of the geometry and the stepper used for tracking the particle can defined in the detector constructor. The physics list is used to define various particles and physical processes. Finally, the primary generator was used to initialize the particle. I utilized the test bed to test out various steppers already present in the codebase to profile them.
```
Comment (agheata) [inline _gsocblogs/2022/blog_Geant4_DivyanshTiwari.md:28]:
```suggestion
Before starting work on the actual methods, it was important to develop a testbed application that could be used to test the steppers and drivers. Hence, I started off my coding period by implementing a geant4 application to track a particle in a geometry initialized with a magnetic field. This was really interesting and exciting for me as I had done my homework during the community bonding period. The testbed is comprised of a detector constructor class, a physics list class, and a primary action generator class. The magnetic field of the geometry and the stepper used for tracking the particle can be defined in the detector constructor. The physics list is used to define various particles and physical processes. Finally, the primary generator was used to initialize the particle. I utilized the test bed to test out various steppers already present in the codebase to profile them.
```

**Review by agheata — CHANGES_REQUESTED** [review]

File: _gsocblogs/2022/blog_Geant4_DivyanshTiwari.md (hunk 1 above)
Comment (divyanshtiwari237) [reply _gsocblogs/2022/blog_Geant4_DivyanshTiwari.md]: Made the requested change

File: _gsocblogs/2022/blog_Geant4_DivyanshTiwari.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+year: 2022
+layout: blog_post
+logo: Geant4-logo.png
+intro: |
+  Hello Everyone! This is Divyansh. I was accepted to take part in GSoC 2022 under CERN-HSF and I am contributing to Geant4. My project is about implementing symplectic integrators into the Geant4 codebase. 
+---
+
+## Introduction
+
+Many applications require the use of integrator method that preserve evergy over the course of a simulation.
```
Comment (agheata) [inline _gsocblogs/2022/blog_Geant4_DivyanshTiwari.md:15]:
```suggestion
Many applications require the use of an integration method that preserves the energy over the course of a simulation.
```

**Review by agheata — CHANGES_REQUESTED** [review]
Minor fixes

**Pull request merged**